src/services/color/color.service.ts

```diff
@@ -8,69 +8,61 @@ import {
 
 export class ColorService {
 
-    private _html = '<div class="primary-color"></div>' +
-    '<div class="accent-color"></div>' +
-    '<div class="secondary-color"></div>' +
-    '<div class="alternate1-color"></div>' +
-    '<div class="alternate2-color"></div>' +
-    '<div class="alternate3-color"></div>' +
-    '<div class="vibrant1-color"></div>' +
-    '<div class="vibrant2-color"></div>' +
-    '<div class="grey1-color"></div>' +
-    '<div class="grey2-color"></div>' +
-    '<div class="grey3-color"></div>' +
-    '<div class="grey4-color"></div>' +
-    '<div class="grey5-color"></div>' +
-    '<div class="grey6-color"></div>' +
-    '<div class="grey7-color"></div>' +
-    '<div class="grey8-color"></div>' +
-    '<div class="chart1-color"></div>' +
-    '<div class="chart2-color"></div>' +
-    '<div class="chart3-color"></div>' +
-    '<div class="chart4-color"></div>' +
-    '<div class="chart5-color"></div>' +
-    '<div class="chart6-color"></div>' +
-    '<div class="ok-color"></div>' +
-    '<div class="warning-color"></div>' +
-    '<div class="critical-color"></div>';
+    private _defaultTheme = [
+        'primary',
+        'accent',
+        'secondary',
+        'alternate1',
+        'alternate2',
+        'alternate3',
+        'vibrant1',
+        'vibrant2',
+        'grey1',
+        'grey2',
+        'grey3',
+        'grey4',
+        'grey5',
+        'grey6',
+        'grey7',
+        'grey8',
+        'chart1',
+        'chart2',
+        'chart3',
+        'chart4',
+        'chart5',
+        'chart6',
+        'ok',
+        'warning',
+        'critical'
+    ];
+
+
+    private _html: string = '';
 
     private _element: HTMLElement;
-    private _colors: any;
+    private _colors: any = {};
+
+    private _theme: string[] = this._defaultTheme;
```

Comment:
Let's establish a proper type for this, even if it is currently just a list of strings. We might want to add more stuff later, I'm thinking that it would be more efficient to precompile the colors in by parsing SCSS, for example.